### PR TITLE
[TE] Adjust dectection window using expected delay

### PR DESCRIPTION
**Pull request by xiaohui-sun**
Add detection expected delay when setting the detection window.
This is useful since some datasets have pretty large delay (e.g, 8 days). Using the default look back window can't catch the data.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineJob.java [hunk 1]
```diff
@@ -56,9 +57,10 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
     Long id = getIdFromJobKey(jobKey.getName());
     DetectionConfigDTO configDTO = detectionDAO.findById(id);
 
-    // Make sure start time is not out of DETECTION_TASK_MAX_LOOKBACK_WINDOW
+    // Get the time window for the detection job.
+    long delay = getDetectionExpectedDelay(configDTO);
     long end = System.currentTimeMillis();
-    long start = Math.max(configDTO.getLastTimestamp(), end  - ThirdEyeUtils.DETECTION_TASK_MAX_LOOKBACK_WINDOW);
+    long start = Math.max(configDTO.getLastTimestamp(), end - ThirdEyeUtils.DETECTION_TASK_MAX_LOOKBACK_WINDOW - delay);
```
Comment (jihaozh) [inline thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineJob.java]: Shall we change it in the `DataAvailabilityTaskScheduler` too?

Comment (xiaohui-sun) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineJob.java]: Good point. I am looking into that.

Comment (vincentchenjl) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineJob.java]: I almost finished this change in `DataAvailabilityTaskScheduler` and am sending a PR on this.

Comment (xiaohui-sun) [reply thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/DetectionPipelineJob.java]: I am thinking of we can simplify this logic. Let's sync offline.

**Pull request merged**